src/coreclr/src/jit/compiler.h

```diff
@@ -3735,26 +3735,21 @@ class Compiler
     void impInsertStmtBefore(Statement* stmt, Statement* stmtBefore);
     Statement* impAppendTree(GenTree* tree, unsigned chkLevel, IL_OFFSETX offset);
     void impInsertTreeBefore(GenTree* tree, IL_OFFSETX offset, Statement* stmtBefore);
-    void impAssignTempGen(unsigned    tmp,
-                          GenTree*    val,
-                          unsigned    curLevel,
-                          Statement** pAfterStmt = nullptr,
-                          IL_OFFSETX  ilOffset   = BAD_IL_OFFSET,
-                          BasicBlock* block      = nullptr);
-    void impAssignTempGen(unsigned             tmpNum,
-                          GenTree*             val,
-                          CORINFO_CLASS_HANDLE structHnd,
-                          unsigned             curLevel,
-                          Statement**          pAfterStmt = nullptr,
-                          IL_OFFSETX           ilOffset   = BAD_IL_OFFSET,
-                          BasicBlock*          block      = nullptr);
+    void impAssignTempGenTree(unsigned tmp, GenTree* val, unsigned curLevel, IL_OFFSETX offset = BAD_IL_OFFSET);
+
+    void impAssignTempGenStruct(unsigned             tmpNum,
+                                GenTree*             val,
+                                CORINFO_CLASS_HANDLE structHnd,
+                                unsigned             curLevel,
+                                Statement**          pAfterStmt = nullptr,
```

Comment:
I have not found an easy way to get rid of `pAfterStmt ` and `block` here, it is used but in very rare cases and these cases are not straightforward.

Reply:
@mikedn don't you accidentally have changes of this code in one of your branches? :-)

Reply:
@sandreenko Not that I remember. I may have something somewhere that replaces the class handle with the class layout thing but it was only an exploratory change. In general I'm trying to keep importer changes to a minimum.